docs/03_features/pydantic.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Pydantic Support
+path: /docs/feature/pydantic
+---
+
+# Pydantic support
+
+Strawberry comes with support for
+[Pydantic](https://pydantic-docs.helpmanual.io/). This allows to create
+Strawberry types from Pydantic models without having to write code twice.
+
+Here's a basic example of how this works, let's say we have a Pydantic Model for
+a user, like this:
+
+```python
+from datetime import datetime
+from typing import List, Optional
+from pydantic import BaseModel
+
+
+class User(BaseModel):
+    id: int
+    name: str
+    signup_ts: Optional[datetime] = None
+    friends: List[int] = []
+```
+
+We can create a Strawberry type by using the `strawberry.beta.pydantic.type`
+decorator:
+
+```python
+import strawberry
+
+from .models import User
+
+@strawberry.beta.pydantic.type(model=User, fields=[
+    'id',
+    'name',
+    'friends'
+])
+class User:
+    pass
+```
+
+The `strawberry.beta.pydantic.type` decorator accepts a Pydantic model and a list of
+fields that we want to expose on our GraphQL API.
+
+> **Note** specifying the list of field is required to prevent accidentally
+> exposing fields that weren't meant to be exposed on a API
+
+## Input types
+
+Input types are similar to types, we can create one by using the
+`strawberry.beta.pydantic.input` decorator:
+
+```python
+import strawberry
+
+from .models import User
+
+@strawberry.beta.pydantic.input(model=User, fields=[
+    'id',
+    'name',
+    'friends'
+])
+class UserInput:
+    pass
+```
+
+## Error Types
+
+In addition to object types and input types strawberry allows you to create
+"error types", you can use these error types to have a typed representation of
+Pydantic errors in GraphQL, let's see an example:
+
+```python
+import pydantic
+import strawberry
+
+class User(BaseModel):
+    id: int
+    name: pydantic.constr(min_length=2)
+    signup_ts: Optional[datetime] = None
+    friends: List[int] = []
+
+@strawberry.beta.pydantic.error_type(model=User, fields=[
+    'id',
+    'name',
+    'friends'
+])
+class UserError:
+    pass
+```
+
+The generated type will look like this:
+
+```python
+@strawberry.type
+class UserError:
+    id: List[str]
+    name: List[str]
+    friends: List[List[str]]
+```
```

Comment:
I think it makes more sense to show the SDL here rather than the equivalent Strawberry type. You can even use the new `python+schema` code block to show them side by side (https://github.com/strawberry-graphql/strawberry.rocks/pull/23)

e.g.
```suggestion
```python+schema
import pydantic
import strawberry

class User(BaseModel):
    id: int
    name: pydantic.constr(min_length=2)
    signup_ts: Optional[datetime] = None
    friends: List[int] = []
    
@strawberry.beta.pydantic.error_type(model=User, fields=[
    'id',
    'name',
    'friends'
])
class UserError:
    pass
---
type UserError {
  id: [String!]!
  name: [String!]!
  friends: [[String!]!]!
}
```
```

Reply:
will change